huaweicloud/services/ga/resource_huaweicloud_ga_address_group.go

```diff
@@ -0,0 +1,424 @@
+		d.Set("listeners", flattenGetListenersResponseBody(getIpAddressGroupRespBody)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func buildUpdateIpAddressGroupBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"ip_group": map[string]interface{}{
+			"name":        utils.ValueIngoreEmpty(d.Get("name")),
+			"description": d.Get("description"),
+		},
+	}
+	return bodyParams
+}
+
+func resourceIpAddressGroupUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	if d.HasChanges("name", "description") {
+		var (
+			updateIpAddressGroupHttpUrl = "v1/ip-groups/{ip_group_id}"
+			updateIpAddressGroupProduct = "ga"
+		)
+		updateIpAddressGroupClient, err := cfg.NewServiceClient(updateIpAddressGroupProduct, region)
+		if err != nil {
+			return diag.Errorf("error creating IP address group Client: %s", err)
+		}
+
+		updateIpAddressGroupPath := updateIpAddressGroupClient.Endpoint + updateIpAddressGroupHttpUrl
+		updateIpAddressGroupPath = strings.ReplaceAll(updateIpAddressGroupPath, "{ip_group_id}", d.Id())
+
+		updateIpAddressGroupOpt := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+			OkCodes: []int{
+				200,
+			},
+		}
+
+		updateIpAddressGroupOpt.JSONBody = utils.RemoveNil(buildUpdateIpAddressGroupBodyParams(d))
+		_, err = updateIpAddressGroupClient.Request("PUT", updateIpAddressGroupPath, &updateIpAddressGroupOpt)
+		if err != nil {
+			return diag.Errorf("error updating IP address group: %s", err)
+		}
+	}
+
+	return resourceIpAddressGroupRead(ctx, d, meta)
+}
+
+func resourceIpAddressGroupDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var (
+		deleteIpAddressGroupHttpUrl = "v1/ip-groups/{ip_group_id}"
+		deleteIpAddressGroupProduct = "ga"
+	)
+	deleteIpAddressGroupClient, err := cfg.NewServiceClient(deleteIpAddressGroupProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating IP address group Client: %s", err)
+	}
+
+	deleteIpAddressGroupPath := deleteIpAddressGroupClient.Endpoint + deleteIpAddressGroupHttpUrl
+	deleteIpAddressGroupPath = strings.ReplaceAll(deleteIpAddressGroupPath, "{ip_group_id}", d.Id())
+
+	deleteIpAddressGroupOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			204,
+		},
+	}
+	_, err = deleteIpAddressGroupClient.Request("DELETE", deleteIpAddressGroupPath, &deleteIpAddressGroupOpt)
+	if err != nil {
+		return diag.Errorf("error deleting IP address group: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"PENDING"},
+		Target:       []string{"DELETED"},
+		Refresh:      waitIpAddressGroupStatusRefreshFunc(d, meta, true),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		Delay:        10 * time.Second,
+		PollInterval: 5 * time.Second,
```

Comment:
`PollInterval` shouldn't be used if the timeout period is less than 10 seconds.

Reply:
done